### PR TITLE
Issue #13970: Update CI to have Maven Version 3.8.8

### PR DESCRIPTION
**Pull request by relentless-pursuit**
#13970

https://github.com/checkstyle/contribution/pull/810

### CODE REVIEW

**Comment by nrmancuso** [comment]
Converting to draft and placing blocked label while we are testing this update

**Comment by relentless-pursuit** [comment]
@nrmancuso, my circle-ci sonar is failing. shall i make any changes in circle-ci/config.yml? or, it can be ignored?

**Comment by nrmancuso** [comment]
> @nrmancuso, my circle-ci sonar is failing. shall i make any changes in circle-ci/config.yml? or, it can be ignored?

You must remove jq install as I did in my example PR.

**Comment by romani** [comment]
Ply fix CI failures https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/48653691/job/kj5it85e6ia6rr35#L114 there are more in others

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,11 +78,10 @@ jobs:
 
   sonarqube:
     docker:
-      - image: checkstyle/jdk-11-groovy-git-mvn:11.0.13__3.0.9__2.25.1__3.6.3
+      - image: checkstyle/jdk-11-groovy-git-mvn:11.0.20.1__2.4.21__2.42.0__3.9.5
 
     steps:
       - checkout
-      - run: apt-get install -y jq
```
Comment (romani) [inline .circleci/config.yml:85]: We do have install at https://github.com/checkstyle/contribution/commit/f059eb492b3ac9d94eaae39c69b55e17c57f9da6#diff-b2fa088093d6886acdf3dc702a7200e08fbcb57f4df0aea745b2ad401ff15fc9R6

**Review by romani — APPROVED** [review]
Ok to meet if ci pass

File: .circleci/config.yml (hunk 1 above)
Comment (relentless-pursuit) [reply .circleci/config.yml:85]: `image-name: &custom_img "checkstyle/jdk-11-groovy-git-mvn:11.0.20.1__2.4.21__2.42.0__3.9.5"` is a line-length violation. should i alter it to allow of greater length or do some hack to accomodate it?

Comment (romani) [reply .circleci/config.yml:85]: Try to shorten line, try `cs_img` instead of custom_img

Comment (relentless-pursuit) [reply .circleci/config.yml]: Sonarqube fails when i use `run: apt-get install -y jq`

Comment (romani) [reply .circleci/config.yml]: Remove
https://github.com/checkstyle/checkstyle/blob/503b26a00b6669775e74a483050c9202487bb00a/.circleci/config.yml#L85

You already have jq in image , no need to install it again

Comment (nrmancuso) [reply .circleci/config.yml:85]: Explanation is at https://github.com/checkstyle/checkstyle/pull/14012#issuecomment-1813681382

**Review by nrmancuso — APPROVED** [review]

**Comment by romani** [comment]
@suniti0804 , can you close your PR to let others finish issue?

**Comment by suniti0804** [comment]
> @suniti0804 , can you close your PR to let others finish issue?

done

**Pull request merged**